trie/database.go

```diff
@@ -535,7 +553,7 @@ func (db *Database) Dereference(root common.Hash) {
 }
 
 // dereference is the private locked version of Dereference.
-func (db *Database) dereference(child common.Hash, parent common.Hash) {
+func (db *Database) dereference(batch ethdb.KeyValueWriter, child common.Hash, parent common.Hash) {
 	// Dereference the parent-child
 	node := db.dirties[parent]
```

Comment:
if `parent` is not in `dirties`, what we do ?

Reply:
It might be out of scope of the PR, because it's this way in the original geth's code. It looks like `parent` always should exist because it's a previous accessed node or zero for root